lib/chef/provider/execute.rb

```diff
@@ -69,6 +69,14 @@ def sensitive?
         !!new_resource.sensitive
       end
 
+      def stream_to_formatter?
+        Chef::Config[:always_stream_execute] || run_context.events.formatter?
```

Comment:
Yeah, also because we have an `elsif` below, this would totally change the behavior of `-l debug` as far as showing command output. I think we should probably rewind and outline the desired semantics in English before working with them as code.